src/main/java/edu/harvard/iq/dataverse/api/Users.java

```diff
@@ -98,4 +101,86 @@ public Response changeAuthenticatedUserIdentifier(@PathParam("identifier") Strin
         return ok("UserIdentifier changed from " + oldIdentifier + " to " + newIdentifier);
     }
     
+    @Path("token")
+    @DELETE
+    public Response deleteToken() {
+        User u;
+
+        try {
+            u = findUserOrDie();
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        }
+        AuthenticatedUser au;        
+       
+        try{
+             au = (AuthenticatedUser) u; 
+        } catch (ClassCastException e){ 
+            //if we have a non-authentivated user we stop here.
+            return notFound("Token for " + u.getIdentifier() + " not found.");
```

Comment:
Couldn't the message here be something like "The guest user cannot have an API token"?

Reply:
I modified the error messages to say that tokens for guests/private url not eligible for deletion or recreation.